### DISCUSSION TITLE
Add proxy option to access

### DISCUSSION THREAD

**Discussion by Harry-Jing** [acheong08/EdgeGPT#190] [Q&A]
Question: Is better set in global or every time create Chatbot instance

**Comment by acheong08** [comment]
I usually set globally via `export all_proxy=socks5://127.0.0.1:1080/`. Trying to mess with different ways which the proxy URL is formatted is a bit annoying. Since httpx, respects environment variables, it should be fine

**Comment by Harry-Jing** [comment]
seem like only https://edgeservices.bing.com/edgesvc/turing/conversation/create ckeck ip

**Comment by PaTTeeL** [comment]
ValueError: Unknown scheme for proxy URL URL('socks5h://ip:port')

Can we support scheme socks5h? As socks5://ip:port requests DNS locally in cURL, and socks5h forces it remotely.

But pip seems works fine.

**Reply by acheong08** [reply]
httpx doesn't support socks5h for some reason

**Reply by PaTTeeL** [reply]
I'm now using command `python3 -m EdgeGPT --cookie-file ~/cookies.json --proxy socks5://ip:port` to launch EdgeGPT.

But there is a environment variable with `declare -x https_proxy="socks5h://ip:port"`. The program crashes with `ValueError: Unknown scheme for proxy URL URL('socks5h://ip:port')` when I'm using !reset in EdgeGPT. Then can we fix it?
